scripts/generate_community_telemetry_figures.py

```
@@ -0,0 +1,265 @@
+    configured = payload["configured_rails"]
+    bucket_total = sum(item["records"] for item in configured["buckets"])
+    if bucket_total != configured["eligible_startup_records"]:
+        raise ValueError("configured rail buckets do not match eligible_startup_records")
+    if configured["excluded_zero_records"] + bucket_total != snapshot["startup_records"]:
+        raise ValueError("configured rail records do not match startup_records")
+    return payload
+
+
+def _snapshot_label(data: dict[str, Any]) -> str:
+    versions = " and ".join(f"v{version}" for version in data["snapshot"]["versions"])
+    return f"{versions} · May 22–Aug 18, 2026"
```

Ruff (0.16.1)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)